### PR TITLE
fix: redirect rmrk2 prefix to ksm

### LINKED ISSUE #5913: When I open rmrk2 item directly with link doesn't load
https://canary.kodadot.xyz/rmrk2/gallery/17842583-22708b368d163c8007-CITY-LOWER_ART_DISTRICT-00000006

image

prob related
- https://github.com/kodadot/nft-gallery/issues/5908

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #5913
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [x] I've tested it at https://deploy-preview-5918--koda-canary.netlify.app/rmrk2/gallery/17842583-22708b368d163c8007-CITY-LOWER_ART_DISTRICT-00000006
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at c907ea7</samp>

Improved middleware for network switching and readability. Added feature flag to redirect `rmrk2` to `ksm` routes in `middleware/redirects.ts`.

### <samp>🤖 Generated by Copilot at c907ea7</samp>

> _`rmrk2` or `ksm`?_
> _Middleware can switch them_
> _A flag cuts the flow_

### CODE REVIEW

**Review by roiLeo — APPROVED** [review]
✅ wfm

File: middleware/redirects.ts [hunk 1]
```diff
@@ -8,4 +9,8 @@ export default function ({ store, redirect, route }): void {
       return redirect(`/${prefix}/explore/items`)
     }
   }
+
+  if (route.path.includes('/rmrk2/')) {
+    return redirect(route.path.replace('/rmrk2/', '/ksm/'))
```
Comment (Jarsen136) [inline middleware/redirects.ts:14]: I would prefer to add the query parameter.

For now, If you open
`https://deploy-preview-5918--koda-canary.netlify.app/rmrk2/explore/items?listed=true&sort=updatedAt_DESC`, it will redirect you to `https://deploy-preview-5918--koda-canary.netlify.app/ksm/explore/items?listed=true` and the query `sort=updatedAt_DESC` would be lost.

**Comment by yangwao** [comment]
question, are there `rmrk2` outside somewhere wild?

File: middleware/redirects.ts (hunk 1 above)
Comment (preschian) [reply middleware/redirects.ts]: > I would prefer to add the query parameter.

oh thanks, updated

**Comment by vikiival** [comment]
> question, are there `rmrk2` outside somewhere wild?

No they shouldn't be.

Still wondering how did you get that link.

**Review by Jarsen136 — APPROVED** [review]

**Comment by preschian** [comment]
> > question, are there `rmrk2` outside somewhere wild?
> 
> No they shouldn't be.
> 
> Still wondering how did you get that link.

found one https://twitter.com/Abdulra49391530/status/1646776000703782914?s=20

**Comment by yangwao** [comment]
ok u win 

pay 20 usd


> Still wondering how did you get that link.

i open links in my quantum state machine

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $20 @ 24.48 USD/KSM ~ 0.817 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0x1d119e697d794cbb35818a584aa73a4843f0c88bab8784efdab13865b9d9ac29](https://kusama.subscan.io/extrinsic/0x1d119e697d794cbb35818a584aa73a4843f0c88bab8784efdab13865b9d9ac29)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)
